web/pdf_viewer.js

```diff
@@ -883,6 +883,15 @@ class PDFViewer {
         // Ensure that the various layers always get the correct initial size,
         // see issue 15795.
         this.viewer.style.setProperty("--scale-factor", viewport.scale);
+        if (window.matchMedia("(forced-colors: active)").matches) {
+          const url = pdfDocument.filterFactory.addHighlightHCMFilter(
+            this.pageColors.foreground,
+            this.pageColors.background,
+            "HighlightText",
+            "Highlight"
+          );
+          this.viewer.style.setProperty("--hcm-highligh-filter-url", url);
+        }
```

Comment:
The feature testing has already been done in the viewer, so I don't think that we need to do that here as well, hence the correct check would be the following instead:
```suggestion

        if (this.pageColors) {
          const url = pdfDocument.filterFactory.addHighlightHCMFilter(
            this.pageColors.foreground,
            this.pageColors.background,
            "HighlightText",
            "Highlight"
          );
          this.viewer.style.setProperty("--hcm-highligh-filter-url", url);
        }
```

Reply:
Please also add the same block [just after this line](https://github.com/mozilla/pdf.js/blob/e9ccbd28561d4c9aa2d666cdd337aa943f44a6f9/web/pdf_page_view.js#L205) as well, for the benefit of the viewer-components.